devito/ir/ietxdsl/cluster_to_ssa.py

```diff
@@ -634,6 +643,8 @@ def apply_timers(module, **kwargs):
     """
     Apply timers to a module
     """
+    if kwargs['xdsl_num_sections'] < 1:
```

Comment:
hm..

Reply:
Long story short
We have a passe adding timer(s) as a float pointer, as that's the mechanism used by Devito (welp, lets it not crash at least :wink:)
Thing is, Devito passes as many floats as it generates "sections", including not passing any pointer if there are no such sections. This is the case in the tested example. So this simply removes it from the function signature!
Otherwise, or call-time code tries to find a pointer to pass to it and crashes :shrug: 

IMO, this is good enough and we should just tidy up the whole call-time side of things too.